core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/DecorrelateInnerUnnestWithGlobalAggregation.java

```diff
@@ -219,8 +219,7 @@ private static boolean isGlobalAggregation(PlanNode node)
         }
 
         AggregationNode aggregationNode = (AggregationNode) node;
-        return aggregationNode.hasEmptyGroupingSet() &&
-                aggregationNode.getGroupingSetCount() == 1 &&
+        return aggregationNode.isSingleGlobalAggregation() &&
                 aggregationNode.getStep() == SINGLE;
```

Comment:
Doing `aggregationNode.isSingleGlobalAggregation()` and then checking `aggregationNode.getStep() == SINGLE` in next line seems redundant. Maybe `isSingleGlobalAggregation()` should be named differently, e.g.:
`hasSingleGlobalAggregation`?

Reply:
`hasSingleGlobalAggregation` sounds good. renamed.